grails-app/views/job/_jobTable.gsp

```diff
@@ -21,7 +21,8 @@
 
             <div class="jobNameContainer"><a
                     href="${createLink(action: 'edit', id: job.id)}"
-                    class="jobName ${job.active == false ? 'inactiveJob' : ''}">${job.label}
+                    class="jobName ${job.active == false ? 'inactiveJob' : ''}">
+                ${!job.label ? job.script.label + ' ' + job.location.browser.name : job.label}
```

Comment:
Can be simplified as `${jobLabel ?: job.script.label + ' ' + job.location.browser.name}`